frontend/src/pages/create_user.tsx

```diff
@@ -42,12 +55,36 @@ export default function CreateUser() {
     }
   };
 
-  const onSubmit: SubmitHandler<FieldValues> = (data) => {
-    setEmailError(false); // add email error logic here
-    console.log(data);
+  const onSubmit: SubmitHandler<FieldValues> = async (data) => {
+    try {
+      setEmailError(false);
+
+      // const emailExists = await checkEmailExists(data.email);
+      const emailExists = await checkEmailExists(String(data.email));
+
+      console.log("Checked Email");
 
-    void router.push("/create_user_2");
+      if (emailExists) {
+        setEmailError(true);
+        return;
+      }
+    } catch (error) {
+      console.error("Error checking if email already in use: ", error);
+    }
+
+    // void router.push("/create_user_2");
+    void router.push({
```

Comment:
Right now I am able to proceed the next create user page without actually filling in the values. There has to be some check to ensure all fields are inputted correctly.

Reply:
Also it correctly detects whether the passwords match but I am still able to go to create user 2 even if the passwords are incorrect. I should only be able to route to create user 2 if all the values are properly filled in.

Reply:
Resolved, but user needs to click out of the "confirm password" box for the button to activate (i.e., click somewhere else on the screen)